test/common/router/config_impl_test.cc

```diff
@@ -2392,17 +2403,15 @@ class RouterMatcherHashPolicyTest : public testing::Test, public ConfigImplTestB
 TEST_F(RouterMatcherHashPolicyTest, HashHeaders) {
   firstRouteHashPolicy()->mutable_header()->set_header_name("foo_header");
   {
-    Http::TestRequestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo", "GET");
-    Router::RouteConstSharedPtr route = config().route(headers, 0);
-    EXPECT_FALSE(route->routeEntry()->hashPolicy()->generateHash(nullptr, headers, add_cookie_nop_,
-                                                                 nullptr));
-  }
-  {
-    Http::TestRequestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo", "GET");
-    headers.addCopy("foo_header", "bar");
-    Router::RouteConstSharedPtr route = config().route(headers, 0);
-    EXPECT_TRUE(route->routeEntry()->hashPolicy()->generateHash(nullptr, headers, add_cookie_nop_,
-                                                                nullptr));
+    EXPECT_FALSE(generateHash({}));
+    EXPECT_TRUE(generateHash({
+        "bar",
+    }));
+
+    absl::Hash<const std::vector<std::string>> hasher;
```

Comment:
I wasn't sure whether `vector<string_view>` and `vector<string>` with the same text would generate the same values. Evidently they do. But maybe it'd be better not to depend on that in the test, so you should use a `vector<string_view>` hasher here.

Also I think these case you have here have no regex-transform.

Can you also add all these cases to your regex ones below? You are not testing all these cases for regex-transform which is basically a separate impl now.